packages/astro/src/integrations/index.ts

```diff
@@ -174,6 +174,20 @@ export async function runHookConfigSetup({
 					updatedSettings.middlewares[order].push(entrypoint);
 				},
 				logger: integrationLogger,
+				injectDts: (dts) => {
+					if (
+						updatedSettings.injectedDts.some(
+							(_dts) => _dts.filename === dts.filename && _dts.source === 'core'
+						)
+					) {
+						logger.warn(
+							null,
+							`The integration ${integration.name} is overriding a core dts: ${dts.filename}`
```

Comment:
I was perfectly ok with the name until I read this error. Should we call them "ambient declarations", which is what they are, instead of "dts", which is a simplification of their extension?

I imagine a user would not know what a "core dts" is. Searching for "core dts" or "dts file" also doesn't help since there is a `.dts` extension. Searching for "ambient declarations" on the other hand immediately directs to TypeScript.

Reply:
I'm open to any naming! All the messages are really WIP and can be improved a lot for sure! (i'll focus on that in a 2nd phase, with documentation) For reference, `@nuxt/kit` has a [`addTypeTemplate` utility](https://github.com/nuxt/nuxt/blob/main/packages/kit/src/template.ts#L39)